Library/Homebrew/cmd/outdated.rb

```diff
@@ -48,6 +48,8 @@ class Outdated < AbstractCommand
 
       sig { override.void }
       def run
+        @greedy = Homebrew::EnvConfig.upgrade_greedy? || args.greedy?
```

Comment:
```suggestion
```

I think if you add `, env: :upgrade_greedy` to `switch "-g", "--greedy"` this will work everywhere you do `args.greedy?` without any other changes being required.